x/wasm/keeper/keeper.go

```diff
@@ -1005,6 +1029,116 @@ func (k Keeper) newQueryHandler(ctx sdk.Context, contractAddress sdk.AccAddress)
 	return NewQueryHandler(ctx, k.wasmVMQueryHandler, contractAddress, k.gasRegister)
 }
 
+func (k Keeper) updateSizeForContract(ctx sdk.Context, contractAddress sdk.AccAddress, sizeChange int64) error {
+	// always charge rent first before updating size, as the charging logic assumes constant size since last
+	// charged block
+	if err := k.chargeRent(ctx, contractAddress); err != nil {
+		return err
+	}
+	sizeKey := types.GetContractStoreSizePrefix(contractAddress)
+	store := ctx.KVStore(k.storeKey)
+	currentSize, err := k.getCurrentSizeForContract(ctx, contractAddress)
+	if err != nil {
+		return types.ErrUpdateContractSize.Wrap(err.Error())
+	}
+	currentSizeSigned := int64(currentSize) + sizeChange
```

Comment:
nit: `newSizeSigned` right? since its the old plus sizeChange? maybe we should call `currentSize` `oldSize` to be clear on old vs new

Reply:
yeah make sense